### PR TITLE
Experts can access a summary and tell if they take care of needs

### PR DESCRIPTION
**Pull request by ArnoHolo**

### CODE REVIEW

File: app/controllers/experts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  before_action :find_expert
+
+  skip_before_action :authenticate_user!
+
+  layout 'experts'
+
+  def diagnosis
+    associations = [visit: [:visitee, :advisor, facility: [:company]], diagnosed_needs: [:selected_assistance_experts]]
+    @diagnosis = Diagnosis.available_for_expert(@expert).includes(associations).find(params[:diagnosis_id])
+    UseCases::UpdateExpertViewedPageAt.perform(diagnosis_id: params[:diagnosis_id].to_i, expert_id: @expert.id)
```
Comment (Apemb) [inline app/controllers/experts_controller.rb]: is the to_i really necessary ?

File: app/models/diagnosis.rb [hunk 2]
```diff
@@ -19,7 +19,11 @@ class Diagnosis < ApplicationRecord
   scope :of_user, (->(user) { joins(:visit).where(visits: { advisor: user }) })
   scope :reverse_chronological, (-> { order(created_at: :desc) })
   scope :in_progress, (-> { where(step: [1..4]) })
```
Comment (Apemb) [inline app/models/diagnosis.rb]: Maybe change that to `[1..(LAST_STEP _1)]` to be consistent with the completed scope

File: spec/controllers/experts_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        end
+      end
+
+      context 'expert does not have access to diagnosis' do
+        it('raises error') { expect { request }.to raise_error ActiveRecord::RecordNotFound }
+      end
+    end
+  end
+
+  describe 'GET #update_status' do
```
Comment (Apemb) [inline spec/controllers/experts_controller_spec.rb]: Patch and not get

File: app/controllers/experts_controller.rb (hunk 1 above)
Comment (ArnoHolo) [reply app/controllers/experts_controller.rb]: For testing purposes, but would work otherwise

**Pull request merged**